ULS macOS IT: Fix test_basic_configuration_log_format

# Description
With the changes made to the pull request  Fix verify-agent-conf for MacOS ULS [#13035](https://github.com/wazuh/wazuh/pull/13035), the `test_basic_configuration_log_format.py:` test fails.

**Build**: https://devel.ci.wazuh.info/view/Tests/job/Test_integration/1347/

Output:
```configure_local_internal_options_module = None
get_configuration = {'apply_to_modules': ['test_basic_configuration_log_format'], 'metadata': {'configuration': 'wazuh_duplicated_macos_co...cos'}}, {'log_format': {'value': 'macos'}}], 'section': 'localfile'}], 'tags': ['test_basic_configuration_log_format']}
configure_environment = None

    @pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
    def test_log_format(configure_local_internal_options_module, get_configuration,
                        configure_environment):
        """Check if Wazuh log format field of logcollector works properly.
    
        Ensure Wazuh component fails in case of invalid values and works properly in case of valid
        log format values.
    
        Raises:
            TimeoutError: If expected callbacks are not generated.
        """
        cfg = get_configuration['metadata']
    
        control_service('stop', daemon=LOGCOLLECTOR_DAEMON)
        truncate_file(LOG_FILE_PATH)
    
        if cfg['valid_value']:
            control_service('start', daemon=LOGCOLLECTOR_DAEMON)
            check_log_format_valid(cfg)
        else:
            if sys.platform == 'win32':
                pytest.xfail("Windows agent allows invalid localfile configuration:\
                              https://github.com/wazuh/wazuh/issues/10890")
                expected_exception = ValueError
            else:
                expected_exception = sb.CalledProcessError
    
            with pytest.raises(expected_exception):
>               control_service('start', daemon=LOGCOLLECTOR_DAEMON)
E               Failed: DID NOT RAISE <class 'subprocess.CalledProcessError'>

test_logcollector/test_configuration/test_basic_configuration_log_format.py:262: Failed
 -----------------------------Captured stdout setup------------------------------ 
Wed Apr  6 00:10:21 UTC 2022

 ------------------------------Captured stderr call------------------------------ 
2022/04/05 17:10:21 wazuh-logcollector[11215] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2022/04/05 17:10:21 wazuh-logcollector[11215] main.c:126 at main(): DEBUG: Wazuh home directory: /Library/Ossec
2022/04/05 17:10:21 wazuh-logcollector[11215] agent_op.c:218 at os_read_agent_profile(): DEBUG: Calling os_read_agent_profile().
2022/04/05 17:10:21 wazuh-logcollector[11215] agent_op.c:237 at os_read_agent_profile(): DEBUG: os_read_agent_profile() = [centos, centos7, centos7.4]
2022/04/05 17:10:21 wazuh-logcollector[11215] config.c:407 at ReadConfig(): DEBUG: agent_config element does not have any attributes.
```

--- a/deps/wazuh_testing/wazuh_testing/logcollector.py
+++ b/deps/wazuh_testing/wazuh_testing/logcollector.py
@@ -378,6 +378,11 @@ def callback_log_stream_exited_error():
     return monitoring.make_callback(pattern=log_format_message, prefix=prefix)
 
 
+def callback_duplicated_macos_log_file():
+    log_format_message = "WARNING: \(\d+\): Log file 'macos' is duplicated."
+    return monitoring.make_callback(pattern=log_format_message, prefix=prefix)
+
+
 def callback_reconnect_eventchannel(location):
     """Create a callback to detect if specified channel has been reconnected successfully.
     Args:

--- a/tests/integration/test_logcollector/test_configuration/data/wazuh_duplicated_macos_configuration.yaml
+++ b/tests/integration/test_logcollector/test_configuration/data/wazuh_duplicated_macos_configuration.yaml
@@ -8,9 +8,9 @@
     - name: 'logcollector_configuration_block1'
     elements:
     - location:
-        value: LOCATION1
+        value: LOCATION
     -  log_format:
-         value: LOG_FORMAT1
+         value: LOG_FORMAT
   - section: localfile
     attributes:
     - name: 'logcollector_configuration_block2'

--- a/tests/integration/test_logcollector/test_configuration/test_basic_configuration_log_format.py
+++ b/tests/integration/test_logcollector/test_configuration/test_basic_configuration_log_format.py
@@ -80,8 +80,8 @@ windows_tcases = [
 
 macos_tcases = [{'LOCATION': 'macos', 'LOG_FORMAT': 'macos', 'VALID_VALUE': True},
                 {'LOCATION': '/tmp/log.txt', 'LOG_FORMAT': 'macos', 'VALID_VALUE': True},
-                {'LOCATION1': 'macos', 'LOG_FORMAT1': 'macos', 'LOCATION2': 'macos', 'LOG_FORMAT2': 'macos',
-                 'VALID_VALUE': False, 'CONFIGURATION': 'wazuh_duplicated_macos_configuration.yaml'},
+                {'LOCATION': 'macos', 'LOG_FORMAT': 'macos', 'LOCATION2': 'macos', 'LOG_FORMAT2': 'macos',
+                 'VALID_VALUE': True, 'CONFIGURATION': 'wazuh_duplicated_macos_configuration.yaml'},
                 {'LOG_FORMAT': 'macos', 'VALID_VALUE': True,
                  'CONFIGURATION': 'wazuh_no_defined_location_macos_configuration.yaml'}
                 ]
@@ -116,7 +116,8 @@ metadata_multiple_logcollector_configuration = [metadata_value for metadata_valu
                                                 'configuration' in metadata_value and
                                                 metadata_value['configuration'] == multiple_logcollector_configuration]
 
-configuration_ids += [f"{x['location1']}_{x['log_format1']}_{x['location1']}_{x['log_format2']}" for x in metadata_multiple_logcollector_configuration]
+configuration_ids += [f"{x['location']}_{x['log_format']}_{x['location2']}_{x['log_format2']}"
+                      for x in metadata_multiple_logcollector_configuration]
 
 configurations += load_wazuh_configurations(configurations_path_multiple_logcollector, __name__,
                                             params=parameters_multiple_logcollector_configuration,
@@ -179,6 +180,10 @@ def check_log_format_valid(cfg):
             log_callback = logcollector.callback_missing_location_macos()
             wazuh_log_monitor.start(timeout=5, callback=log_callback,
                                     error_message="The expected warning missing location value has not been produced")
+        if 'location2' in cfg and cfg['location2'] == 'macos':
+            log_callback = logcollector.callback_duplicated_macos_log_file()
+            wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                    error_message="The expected warning Log file 'macos' is duplicated")
 
         log_callback = logcollector.callback_monitoring_macos_logs()
         wazuh_log_monitor.start(timeout=5, callback=log_callback,

